### PR TITLE
Cookie Authentication and Data Seeding Actors

### PR DESCRIPTION
**Pull request by wintondeshong**
**Additions**
- Environment and HTTP constants
- Cookie and Seed configuration classes
- Base seed abstract class (more of this abstraction to come)
- EncryptionUtils (used by auth currently)

**Checklist**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [-] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/AndcultureCode.CSharp.Core/Models/Configuration/CookieAuthenticationConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿namespace AndcultureCode.CSharp.Business.Core.Models.Configuration
+{
+    /// <summary>
+    /// Cookie authentication settings
+    /// </summary>
+    public class CookieAuthenticationConfiguration
+    {
+        /// <summary>
+        /// Where to redirect user when access is denied.
+        /// Only applies to tranditional web view auth flows.
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Core/Models/Configuration/CookieAuthenticationConfiguration.cs:10]: Minor: typo, `tranditional`.

File: src/AndcultureCode.CSharp.Core/Models/Configuration/CookieAuthenticationConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        public string CookieName { get; set; }
+
+        /// <summary>
+        /// Is cookie authentication currently enabled?
+        /// </summary>
+        public bool IsEnabled { get; set; }
+
+        /// <summary>
+        /// Where to redirect the user when they are required to login.
+        /// Only applies to tranditional web view auth flows.
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Core/Models/Configuration/CookieAuthenticationConfiguration.cs:31]: Same as above.

File: test/AndcultureCode.CSharp.Core.Tests/Utilities/Security/EncryptionUtilsTest.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        [Fact]
+        public void GenerateSalt_When_Bits_IsNot_DivisibleBy_8_Then_Throws_Exception() =>
+            Assert.Throws<ArgumentException>(() =>
+            {
+                EncryptionUtils.GenerateSalt(
+                    bits: 129 // <----- not divisible by 8, but greater than minimum of 128
+                );
+            });
+
+
```
Comment (Stefanie899) [inline test/AndcultureCode.CSharp.Core.Tests/Utilities/Security/EncryptionUtilsTest.cs:90]: Super minor, but there's additional spacing here compared to the rest of the file.

File: test/AndcultureCode.CSharp.Core.Tests/Utilities/Security/EncryptionUtilsTest.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        [Fact]
+        public void GenerateSalt_When_Default_Arguments_Returns_Salt() =>
```
Comment (Stefanie899) [inline test/AndcultureCode.CSharp.Core.Tests/Utilities/Security/EncryptionUtilsTest.cs:93]: Should we be asserting `ShouldBeBase64()` here as well?

Comment (wintondeshong) [reply test/AndcultureCode.CSharp.Core.Tests/Utilities/Security/EncryptionUtilsTest.cs:93]: Good call!

**Comment by wintondeshong** [comment]
Included in version 0.4.8

**Pull request merged**